### PR TITLE
MAT-8956 prevent deleting measure when locked

### PR DESCRIPTION
**Pull request by adongare**
## MADiE PR

Jira Ticket: [MAT-8956](https://jira.cms.gov/browse/MAT-8956)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/MeasureService.java [hunk 1]
```diff
@@ -323,23 +324,48 @@ public Measure updateMeasure(
...
 
+    if (appConfigService.isFlagEnabled(MadieFeatureFlag.LOCKING)) {
+      LockInfo lockInfo = measureLockService.lockMeasure(id, username);
+      if (lockInfo.isLocked() && !username.equals(lockInfo.getLockedBy())) {
+        log.info(
+          "user: [{}] can't de-activate Measure: [{}], because measure is locked by: [{}]",
+          username,
+          id,
+          lockInfo.getLockedBy());
+        throw new LockNotObtainedException(
```
Comment (RohitKandimalla) [inline src/main/java/cms/gov/madie/measure/services/MeasureService.java:353]: I understand that you are using an existing exception, but looks like LockNotObtainedException returns 409, I found that http has a seperate status to indicate locked status

LOCKED(423, HttpStatus.Series.CLIENT_ERROR, "Locked"), May be we could use that

**Review by RohitKandimalla — DISMISSED** [review]

File: src/main/java/cms/gov/madie/measure/services/MeasureService.java (hunk 1 above)
Comment (adongare) [reply src/main/java/cms/gov/madie/measure/services/MeasureService.java:353]: yeah locked seems more logical. updated it.

**Review by sb-cecilialiu — APPROVED** [review]
Very nice! Thanks!!
